File: 3 - C++/C++ - Course 2 - Classes/Round 3/Osoite.cpp
#include "Osoite.h"
#include <iostream>
using namespace std;

Osoite::Osoite() : mKatuosoite("N/A"), mPostinumero("N/A"), mKunta("N/A")
{
	cout << "Osoite -luokan oletusrakentaja (parametriton rakentaja)\n";
}

Osoite::Osoite(string aKatuosoite, string aPostinumero, string aKunta)
	: mKatuosoite(aKatuosoite), mPostinumero(aPostinumero), mKunta(aKunta)
{
	cout << "Osoite -luoka 3 parametrinen rakentaja\n";
}

Osoite::~Osoite()
{
	cout << "Osoite -luokan purkaja\n";
}

string Osoite::getKatuosoite() const
{
	return mKatuosoite;
}

string Osoite::getPostinumero() const
{
	return mPostinumero;
}
string Osoite::getKunta() const
{
	return mKunta;
}

void Osoite::setKatuosoite(string aKatuosoite)
{
	mKatuosoite = aKatuosoite;
}

void Osoite::setPostinumero(string aPostinumero)
{
	mPostinumero = aPostinumero;
}

void Osoite::setKunta(string aKunta)
{
	mKunta = aKunta;
}

void Osoite::tulostaTiedot() const
{
	cout << "Katuosoite: " << mKatuosoite << endl;
	cout << "Postinumero: " << mPostinumero << endl;
	cout << "Kunta: " << mKunta << endl;
}

File: 5 - Ohjelmoinnin syventavat tekniikat/Viikko 2 - 4/Osoite.java
public class Osoite{
    private String mKatuosoite;
    private String mPostinumero;
    private String mKunta;

    private Opiskelija a = new Opiskelija();

    public Osoite(){
        mKatuosoite = "N/A";
        mPostinumero = "N/A";
        mKunta = "N/A";
    }
    public Osoite(String aKatuosoite, String aPostinumero, String aKunta){
        mKatuosoite=aKatuosoite;
        mPostinumero = aPostinumero;
        mKunta = aKunta;
    }


    public void setKatuosoite(String aKatuosoite){
        mKatuosoite = aKatuosoite;
    }
    public void setPostinumero(String aPostinumero){
        mPostinumero = aPostinumero;
    }
    public void setKunta(String aKunta){
        mKunta = aKunta;
    }
    public String getKatuosoite(){
        return mKatuosoite;
    }

    public String getPostinumero(){
        return mPostinumero;
    }
    public String getKunta(){
        return mKunta;
    }

    public void tulostaTiedot(){
        System.out.println("OSOITETIEDOT");
        System.out.println(getKatuosoite());
        System.out.println(getPostinumero() + " " + getKunta());
        System.out.println();
    }
}
